modules/client/src/allowlistVoting/internal/methods.ts

```diff
@@ -0,0 +1,470 @@
+  ContractReceipt,
+  Event as EthersEvent,
+} from "@ethersproject/contracts";
+import { AllowlistVoting } from "../client";
+import {
+  ICreateProposalParams,
+  ISetConfigurationParams,
+  IVoteParams,
+  Steps,
+  Proposal,
+  ProposalCreationValue,
+  VoteStepsValue,
+} from "../interfaces";
+import { arrayify } from "@ethersproject/bytes";
+
+export class AllowlistVotingMethods {
+  private allowlistVoting: AllowlistVoting;
+
+  constructor(allowlistVoting: AllowlistVoting) {
+    this.allowlistVoting = allowlistVoting;
+  }
+
+  /**
+   * Returns the hash for the MODIFY_ALLOWLIST_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public MODIFY_ALLOWLIST_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .MODIFY_ALLOWLIST_PERMISSION_ID();
+  }
+
+  /**
+   * Returns the PCT_BASE
+   *
+   * @return {*}  {Promise<number>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async PCT_BASE(): Promise<number> {
+    return (
+      await this.allowlistVoting.getPluginInstanceWithSigner().PCT_BASE()
+    ).toNumber();
+  }
+
+  /**
+   * Returns the hash for the SET_CONFIGURATION_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public SET_CONFIGURATION_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .SET_CONFIGURATION_PERMISSION_ID();
+  }
+
+  /**
+   * Returns the hash for the UPGRADE_PLUGIN_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public UPGRADE_PLUGIN_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .UPGRADE_PLUGIN_PERMISSION_ID();
+  }
+
+  /**
+   * Adds new allowed users to the plugin.
+   * Returns a generator with 2 steps
+   *
+   * @param {string[]} _users
+   * @return {*}  {AsyncGenerator<VoteStepsValue>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async *addAllowedUsers(
+    _users: string[]
+  ): AsyncGenerator<VoteStepsValue> {
+    const tx = await this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .addAllowedUsers(_users);
+    yield {
+      key: Steps.PENDING,
+      txHash: tx.hash,
+    };
+    await tx.wait();
+    yield {
+      key: Steps.DONE,
+    };
+  }
+
+  /**
+   * Returns the amount of allowed users at a given block
+   *
+   * @param {BigInt} blockNumber
+   * @return {*}  {Promise<BigInt | number>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async allowedUserCount(blockNumber: BigInt | number): Promise<BigInt> {
+    return (
+      await this.allowlistVoting
+        .getPluginInstanceWithSigner()
+        .allowedUserCount(BigNumber.from(blockNumber))
+    ).toBigInt();
+  }
+
+  /**
+   * Checks if a proposal can be executed
+   *
+   * @param {BigInt | number} _proposalId
```

Comment:
Why is `proposalId` a `Bignumber`? 

I guess it is the proposal number for this particular `pluginAddress`, right now we've been calling `proposalId` to the id that subgraph returned `{pluginAddress}_0x{proposal_number}`